catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/content/data/impl/ContentItemValidator.java

```diff
@@ -26,35 +24,52 @@
  * required by the {@link ddf.catalog.content.StorageProvider}.
  */
 public class ContentItemValidator {
-
     private static final Pattern QUALIFIER_PATTERN = Pattern.compile("\\w[a-zA-Z0-9_\\-]+");
```

Comment:
I had forgotten that we strip dashes from the UUID. (as an aside we only do that due to a solr limitation) With that in mind, I see the need for regexes instead.

I'm not sure that we really note this anywhere, or what the correct place to do so would even be. This validator seems like it could use an explanatory comment. As you noted, more naive interpretations have already caused us problems in the past.